vaadin-charts-flow-parent/vaadin-charts-flow-svg-generator/jsdom-exporter.js

```diff
@@ -138,6 +200,67 @@ doc.createElementNS = (ns, tagName) => {
             height: retHeight
         };
     };
+    /**
+     * Estimate the rendered length of a substring of text. Uses a similar strategy to getBBox,
+     * above.
+     *
+     * @param {integer} charnum Starting character position
+     * @param {integer} numchars Number of characters to count
+     * @returns Rendered length of the substring (estimated)
+     */
+    elem.getSubStringLength = (charnum, numchars) => {
+        let offset = charnum,
+            remaining = numchars,
+            textLength = 0;
+
+        let children = [].slice.call(
+            elem.children.length ? elem.children : [elem]
+        );
+
+        children.forEach(child => {
+            if (child.getAttribute('class') === 'highcharts-text-outline') {
+                child.parentNode.removeChild(child);
+            }
+        });
+
+        processTextNodes(elem, (textNode, child, parents) => {
+            if (child.tagName === 'title') {
+                return;
+            }
+
+            if (remaining <= 0) {
+                return;
+            }
+            let childLength = textNode.length;
+
+            if (childLength <= offset) {
+                offset -= childLength;
```

Comment:
None of the current tests seems to hit this condition. Adding at least one test where this condition happens would be a good idea.

Reply:
I don't know of a good way to trigger this case through Highcharts chart generation, so I added a mocha test to directly invoke it. I changed the way the shims were added to elements to enable that testing (and I think it's a neater method, anyway).

Reply:
Cool. I actually had this same refactor in mind while I was reviewing this code. Patching the prototype seems to be a better alternative to what is currently being done.